Dataviz: Prevent chart dropdowns causing horizontal scroll

tkohr opened the pull request:
PR prevents `dropdown-selector` to cause horizontal scroll, while staying big enough to display (at least parts of) the selected value. This is in particular necessary for the `chart-view.component` where dropdowns are displayed without wrapping.

Before:

![before](https://user-images.githubusercontent.com/6329425/228291656-2135e830-3a5a-4292-aab7-917d19b14a5b.png)

After:

![after](https://user-images.githubusercontent.com/6329425/228291718-3bece9e5-24d6-46c1-abe3-5041fe596827.png)

jahow commented on libs/ui/inputs/src/lib/dropdown-selector/dropdown-selector.component.html:
```diff
@@ -1,15 +1,15 @@
   <label
     *ngIf="showTitle"
-    class="block tracking-wide text-sm mb-2 sm:mb-0 sm:mr-2 whitespace-no-wrap"
+    class="block tracking-wide text-sm mb-2 sm:mb-0 sm:mx-2 whitespace-no-wrap shrink-0"
```
ok for the `shrink-0` though!

tkohr replied:
IMO, a spacing on the left here makes sense in general to not have the label stick to the left. Made it a padding to be more meaningful.

jahow replied:
ok, actually I think you're right, a padding on the left makes sense as well!

jahow commented on libs/ui/inputs/src/lib/dropdown-selector/dropdown-selector.component.html:
```diff
@@ -1,15 +1,15 @@
     [id]="id"
     (change)="this.selectValue.emit($event.target.value)"
-    class="text-main bg-white border border-gray-300 py-[10px] px-[16px] rounded-[0.25em] cursor-pointer leading-tight focus:outline-none hover:text-primary-darker hover:border-primary-lighter focus:ring-primary-lightest focus:ring-4"
+    class="w-full max-w-xs text-main bg-white border border-gray-300 py-[10px] px-[16px] rounded-[0.25em] cursor-pointer leading-tight focus:outline-none hover:text-primary-darker hover:border-primary-lighter focus:ring-primary-lightest focus:ring-4"
```
Ok for `w-full` indeed it helps making the dropdown shrink when there's not enough room. But why add `max-w-xs`?

jahow reviewed (commented):
I think part of this fix at least should take place in the datahub app. The presentation components are very low level building blocks of geonetwork-ui and as such must be modified as less as possible, because there's a high probability of breaking app layouts.

To be safe, presentation components must make absolutely no assumption on what's around them, and simply fit as best as they can in the box they are given. For example, the dropdown selector will switch to a column layout when on a narrow screen but this could actually be improved, since the size of the screen is not a direct indication of how much space it has.

What do you think?

fgravin commented:
"Une image vaut mieux qu'un long discours" 
I don't understand what is the issue while reading :smile: 
nvm @jahow is handling this.

Thanks @tkohr

jahow commented:
I added a resizeable frame in storybook for the dropdown.

Without the left margin/padding:
![image](https://user-images.githubusercontent.com/10629150/228525120-26d6d6f3-f1dc-4480-880f-b52b8cfeb02a.png)

With:
![image](https://user-images.githubusercontent.com/10629150/228525307-489843ab-7935-45aa-8332-b0c585cb5840.png)

The first one seems more balanced.

Also the label may provoke horizontal scrolling as well since it's not shrinkable:
![image](https://user-images.githubusercontent.com/10629150/228525836-024867ff-522d-4571-a624-26faf7ce6a92.png)

So I'm not sure about this change overall... I think both the label and the dropdown should shrink at some point. Having a min width on the dropdown makes sense though, so that i can at least display one letter.

(this is from the chart-view storybook)

tkohr commented:
Thanks for your careful feedback @jahow ! I agree about the left margin, so I moved the left spacing into the `chart-view.component`. Also distributed the space for the different dropdown more evenly here and switched to `flex-col` on small screens. 

Didn't really find a solution to make the dropdown label shrink in a generic way and still displaying the inputs properly next to them. So it is currently up to the parent to make the dropdowns wrap or display in columns if necessary. Hoping this is acceptable.

![dropdowns](https://user-images.githubusercontent.com/6329425/228845209-b7ad8f42-8e2c-4ee2-ad3f-be4a64bd3ebb.png)

jahow reviewed (approved):
Thanks for your work on this!
